src/components/infinite-data-table.jsx

```diff
@@ -0,0 +1,427 @@
+  }
+
+  componentDidUpdate(prevProps) {
+    const { data, columns } = this.props;
+    const { data: prevData } = prevProps;
+    if (!serializableDeepAreEqual(data, prevData)) {
+      for (let rowIndex = prevData.length; rowIndex <= data.length + 1; rowIndex += 1) {
+        for (let columnIndex = 0; columnIndex < columns.length; columnIndex += 1) {
+          this.cache.clear(rowIndex, columnIndex);
+        }
+      }
+    }
+  }
+
+  getHeader() {
+    const { columns } = this.props;
+    return columns.map(column => ({ label: column.label || '', render: x => x }));
+  }
+
+  getColumnWidth({ index }) {
+    const { columns } = this.props;
+    const column = columns[index];
+    return column && column.width ? column.width : DEFAULT_WIDTH;
+  }
+
+  static getSortableHeaderCell({ column, onHeaderClick, style }) {
+    let className = 'table-head ';
+    if (column.sorted) {
+      className += column.sorted === 'ascend' ? 'table-header-ascend' : 'table-header-descend';
+    } else {
+      className += 'table-header-sortable';
+    }
+    return (
+      <button
+        style={style}
+        type="button"
+        key={column.name}
+        className={className}
+        onClick={() => onHeaderClick(column.name)}
+      >
+        {column.label || ''}
+      </button>
+    );
+  }
+
+  static getHeaderCell({ column, onHeaderClick, style }) {
+    if (column.sortable) {
+      return InfiniteDataTableCore.getSortableHeaderCell({ column, onHeaderClick, style });
+    }
+    return (
+      <div className="table-head table-header" style={style}>
+        {column.label || ''}
+      </div>
+    );
+  }
+
+  static getLoadingCell({ className, style }) {
+    return (
+      <div className={className} style={style}>
+        Loading...
+      </div>
+    );
+  }
+
+  static getSelectCell({
+    id, onSelect, style, className, selectedRows,
+  }) {
+    const checked = id in selectedRows;
+    return (
+      <div className={`${className} ${checked && 'table-row-selected'}`} style={style}>
+        <input type="checkbox" onChange={() => onSelect(id)} checked={id in selectedRows} />
+      </div>
+    );
+  }
+
+  static getNumberCell({
+    style, className, rowIndex, showHeader,
+  }) {
+    return (
+      <div className={className} style={style}>
+        {rowIndex + 1 - showHeader}
+      </div>
+    );
+  }
+
+  static getNoRenderCell({ style, column }) {
+    return (
+      <div className="table-data table-row-warning" style={style}>
+        {`${column.name} has no render method`}
+      </div>
+    );
+  }
+
+  static getCell({
+    column,
+    row,
+    style,
+    rowIndex,
+    showHeader,
+    idKey,
+    onSelect,
+    onHeaderClick,
+    selectedRows,
+  }) {
+    if (showHeader && rowIndex === 0) {
+      return InfiniteDataTableCore.getHeaderCell({
+        column,
+        onHeaderClick,
+        style,
+      });
+    }
+
+    let className = `table-data ${rowIndex % 2 ? 'table-row-odd' : 'table-row-even'}`;
```

Comment:
BEM convention for modifiers is `--` eg. `table-row--odd`

Reply:
Done.